### PR TITLE
Feature/backend/authorization

### PR DESCRIPTION
**Pull request by espandre**
## Related Issue
Closes `https://trello.com/c/IfMfAa2o`

## Proposed changes
- Adds backend support for authorization / authentication of users.

## Additional info
- Users need to be manually added to the database at this point. Later issues may open a private endpoint for adding users.
- Added 'jsonwebtoken' dependency.

## How has this been tested?
- [x] Manual post-ing of known data to database.

## Checklist
- [x] Tests
- [x] Documentation

## Screenshots (if appropriate):

### CODE REVIEW

**Comment by espandre** [comment]
Echoing @fredrbus: "Tests were expected to fail, as the ESLint/Prettier bugfix is not implemented in this branch"

File: backend/src/auth/credentials.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+/* Checks if a password matches a provided hash, returning true if it did.
+ * 'password' *must* be a utf-8 encoded string, whilst 'existing_hash' *must* be an ascii encoded string.
+ */
+export const checkPassword = (password: string, existing_hash: string) => {
+  // This is *meant* to be slow. In fact, the slowness is built into the algorithm.
+  // The reason this is slow is because it is used in order to reduce the efficacy of
+  // brute force attacks. The slowness is controlled by the 'rounds' parameter.
+
+  const decoded_hash = decodePasswordHash(existing_hash);
```
Comment (fredrbus) [inline backend/src/auth/credentials.ts:14]: Snake case in TypeScript will be a problem later!

**Review by fredrbus — CHANGES_REQUESTED** [review]
Great work!
I'd suggest that you merge the main branch into this one, and run `yarn run lint-fix` before making a new commit. This way we avoid pushing unlintable code into main, which will be a pain in the ass for everybody later 😅

File: backend/src/auth/credentials.ts (hunk 1 above)
Comment (espandre) [reply backend/src/auth/credentials.ts:14]: Thanks for catching that. It seems like I'm just too used to snake case...

Comment (espandre) [reply backend/src/auth/credentials.ts:14]: We need to relax some of the lint-rules, though. I need bitwise operations in order to implement password comparisons securely.

Comment (espandre) [reply backend/src/auth/credentials.ts:14]: I'm doing a somewhat dirty per line fix for this by disabling the eslint rule on the lines that use bitwise ops, which might be enough.

Comment (espandre) [reply backend/src/auth/credentials.ts]: It should be fixed now, as eslint passes without error on the backend for me. I haven't touched the backend, which is why the CI fails.

**Review by fredrbus — APPROVED** [review]
👌🏼

**Pull request merged**